uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/GeneralPathImpl.java

```diff
@@ -42,7 +42,7 @@ private GeneralPathImpl(final FileSystem fs,
                             boolean isNormalized) {
         super(fs,
               path,
-              "",
+              isRoot ? "master@localhost" : "",
```

Comment:
@ederign @rhtevan  I'm not quite sure about this... I mean yes, we use JGit as our default file system, but "master@localhost" shouldn't be there as this is a GeneralPathImpl. This should still valid in case we use a SimpleFileSystem with this kind of configuration? Maybe I'm missing something here

Reply:
master@localhost doesn't make much sense if our goal is to keep VFS agnostic. @rhtevan could you please refactor this?

Reply:
@adrielparedes @ederign Good eye on this which I do agree it's a bit of odd. It turns out to be a workaround to another hardcoded [scenario](https://github.com/kiegroup/appformer/blob/92d05f8620fb775a9fdd96574273de2deda3d215/uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/config/ConfigurationServiceImpl.java#L129%20*/%20assertThat(root.toUri().toString().contains(%22/master@%22)).isTrue();%20assertThat(root).isEqualTo(fileSystem.getPath(%22/path%22).getRoot());%20assertThat(root).isEqualTo(fileSystem.getPath(%22/path%22).getParent());%20assertThat(root.getRoot().equals(root)).isTrue();%20assertThat(root.toString().equals(%22/%22)).isTrue();%20assertThat(root.toRealPath().toString().equals(%22/%22)).isTrue();%20assertThat(root.getParent()).isNull();%20assertThat(root.getFileName()).isNull();%20assertThat(root.getNameCount()).isEqualTo(0);%20assertThat(root.iterator().hasNext()).isEqualTo(false);%20assertThat(ne.size()).isEqualTo(0);%20assertThat(getFileNameString(root).equals(%22/%22)).isTrue();%20}%20@Test%20public%20void%20testInitRoot()%20{%20final%20FileSystem%20fs%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20final%20Path%20root%20=%20fs.getPath(%22/%22);%20final%20Path%20testParentDir%20=%20fs.getPath(%22/.testParentDir%22);%20final%20Path%20testDir%20=%20fs.getPath(%22/.testParentDir/.testInitRoot%22);%20final%20Path%20rPath%20=%20fs.getPath(%22./../.testInitRoot%22);%20assertThat(root.getParent()).isNull();%20assertThat(root.isAbsolute()).isTrue();%20assertThat(rPath.isAbsolute()).isFalse();%20assertThat(rPath.getRoot()).isNull();%20assertThat(testDir.isAbsolute()).isTrue();%20assertThat(testDir.getParent()).isEqualTo(testParentDir);%20assertThat(testDir.getRoot()).isEqualTo(root);%20assertThat(testDir.getFileName()).isEqualTo(rPath.getFileName());%20Path%20aPath%20=%20((K8SFileSystemProvider)fsProvider).toAbsoluteRealPath(rPath);%20assertThat(aPath.getRoot()).isEqualTo(root);%20assertThat(aPath.getParent()).isNotNull();%20assertThat(aPath.isAbsolute()).isTrue();%20assertThat(aPath.getFileName()).isEqualTo(rPath.getFileName());%20assertThat(testDir.isAbsolute()).isTrue();%20assertThat(testDir.getParent()).isEqualTo(testParentDir);%20assertThat(testParentDir.getParent()).isEqualTo(root);%20assertThat(testParentDir).isEqualTo(((K8SFileSystemProvider)fsProvider).toAbsoluteRealPath(testParentDir));%20CLIENT_FACTORY.get()%20.configMaps()%20.inNamespace(TEST_NAMESPACE)%20.createOrReplace(CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-dir-r-empty-configmap.yml%22))%20.get());%20assertThat(Files.exists(root)).isTrue();%20try%20(DirectoryStream%3CPath%3E%20stream%20=%20Files.newDirectoryStream(root))%20{%20ArrayList%3CPath%3E%20dirContent%20=%20Lists.newArrayList(stream);%20assertThat(dirContent).asList().isEmpty();%20}%20CLIENT_FACTORY.get().configMaps().inNamespace(TEST_NAMESPACE).delete();%20assertThat(Files.exists(root)).isFalse();%20try%20(DirectoryStream%3CPath%3E%20stream%20=%20Files.newDirectoryStream(root))%20{%20ArrayList%3CPath%3E%20dirContent%20=%20Lists.newArrayList(stream);%20assertThat(dirContent).asList().isEmpty();%20}%20assertThat(Files.exists(root)).isTrue();%20assertThat(getFsObjCM(CLIENT_FACTORY.get(),%20root).getData()).isNotNull();%20Files.createDirectory(testDir);%20try%20(DirectoryStream%3CPath%3E%20stream%20=%20Files.newDirectoryStream(testParentDir))%20{%20ArrayList%3CPath%3E%20dirContent%20=%20Lists.newArrayList(stream);%20assertThat(dirContent).asList().containsExactly(testDir);%20}%20}%20@Test(expected%20=%20IllegalStateException.class)%20public%20void%20testAmbiguousCM()%20{%20final%20FileSystem%20fileSystem%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20final%20Path%20file%20=%20fileSystem.getPath(%22/testDir/testFile%22);%20CLIENT_FACTORY.get()%20.configMaps()%20.inNamespace(TEST_NAMESPACE)%20.createOrReplace(CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-file-configmap.yml%22))%20.get());%20CLIENT_FACTORY.get()%20.configMaps()%20.inNamespace(TEST_NAMESPACE)%20.createOrReplace(CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-file-dup-configmap.yml%22))%20.get());%20getFsObjCM(CLIENT_FACTORY.get(),%20file);%20}%20@Test(expected%20=%20IllegalArgumentException.class)%20public%20void%20testInvalidFsObjCM()%20{%20final%20FileSystem%20fileSystem%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20final%20ConfigMap%20cm%20=%20CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-file-invalid-configmap.yml%22))%20.get();%20getPathByFsObjCM((K8SFileSystem)fileSystem,%20cm);%20}%20@Test%20public%20void%20testFileStore()%20{%20final%20K8SFileSystem%20fs%20=%20(K8SFileSystem)fsProvider.getFileSystem(URI.create(%22default:///%22));%20fs.lock();%20fs.unlock();%20assertThat(fs).isNotNull();%20}%20@Test(expected%20=%20UnsupportedOperationException.class)%20public%20void%20testFileStoreGetTotalSpace()%20{%20fstore.getTotalSpace();%20}%20@Test(expected%20=%20UnsupportedOperationException.class)%20public%20void%20testFileStoreGetUsableSpace()%20{%20fstore.getUsableSpace();%20}%20@Test(expected%20=%20UnsupportedOperationException.class)%20public%20void%20testFileStoreGetUnallocatedSpace()%20{%20fstore.getUnallocatedSpace();%20}%20@Test(expected%20=%20UnsupportedOperationException.class)%20public%20void%20testNewFileChannel()%20{%20final%20FileSystem%20fs%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20fsProvider.newFileChannel(fs.getPath(%22/%22),%20null);%20}%20@Test%20public%20void%20testGetFileStore()%20{%20final%20FileSystem%20fs%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20FileStore%20fstore%20=%20fsProvider.getFileStore(fs.getPath(%22/%22));%20assertThat(K8SFileStore.class.isInstance(fstore)).isTrue();%20}%20@Test(expected%20=%20FileAlreadyExistsException.class)%20public%20void%20testCheckFileExistsThenThrow()%20{%20final%20FileSystem%20fileSystem%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20final%20Path%20dir%20=%20fileSystem.getPath(%22/testDir%22);%20CLIENT_FACTORY.get()%20.configMaps()%20.inNamespace(TEST_NAMESPACE)%20.createOrReplace(CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-dir-0-configmap.yml%22))%20.get());%20fsProvider.createDirectory(dir);%20}%20@Test(expected%20=%20UnsupportedOperationException.class)%20public%20void%20testCopyDirShouldFail()%20{%20final%20FileSystem%20fileSystem%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20final%20Path%20source%20=%20fileSystem.getPath(%22/testDir%22);%20final%20Path%20target%20=%20fileSystem.getPath(%22/testDir2%22);%20CLIENT_FACTORY.get()%20.configMaps()%20.inNamespace(TEST_NAMESPACE)%20.createOrReplace(CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-dir-0-configmap.yml%22))%20.get());%20fsProvider.copy(source,%20target);%20}%20@Test%20public%20void%20testReadAttributes()%20{%20final%20FileSystem%20fileSystem%20=%20fsProvider.getFileSystem(URI.create(%22default:///%22));%20final%20Path%20dir%20=%20fileSystem.getPath(%22/testDir%22);%20CLIENT_FACTORY.get()%20.configMaps()%20.inNamespace(TEST_NAMESPACE)%20.createOrReplace(CLIENT_FACTORY.get().configMaps()%20.load(K8SFileSystemInitTest.class.getResourceAsStream(%22/test-k8sfs-dir-0-configmap.yml%22))%20.get());%20assertThat(fsProvider.readAttributes(dir,%20HiddenAttributes.class)).isNull();%20assertThat(fsProvider.readAttributes(dir,%20BasicFileAttributes.class)).isNotNull();%20assertThat(fsProvider.readAttributes(dir,%20BasicFileAttributes.class).isDirectory()).isTrue();%20assertThat(fsProvider.readAttributes(dir,%20BasicFileAttributes.class).isRegularFile()).isFalse();%20assertThat(fsProvider.readAttributes(dir,%20BasicFileAttributes.class).isSymbolicLink()).isFalse();%20assertThat(fsProvider.readAttributes(dir,%20BasicFileAttributes.class).size()).isEqualTo(19);%20}%20}).

The following test code reflects [such](https://github.com/rhtevan/appformer/blob/c1fff372aea4f0b0d134d1355b054f6b29f04beb/uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-k8s/src/test/java/org/uberfire/java/nio/fs/k8s/K8SFileSystemInitTest.java#L93-L97).

Fixing the hardcoding from the original place would be ideal but not sure the impact. Current solution could be pragmatic approach if you would agree.

Reply:
@ederign @adrielparedes What do you thing? Could we leave it as-is for now, or there is a better solution? Thanks!

Reply:
Makes sense, Adriel, could please include this fix on AF-2253 work?

Reply:
of course! In that case when the thread name is set I will approve the PR. Thanks @rhtevan

Reply:
@adrielparedes watcher thread name being added. Thanks